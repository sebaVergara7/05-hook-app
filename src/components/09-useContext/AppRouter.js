import React from "react";
import {
    BrowserRouter as Router,
    Routes,
    // Switch,
    Route,
    Navigate,
    // Redirect
} from "react-router-dom";

import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { NavBar } from "./NavBar";

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />

                <div className='container'>
                    <Routes>
                    {/* <Switch> */}
                        <Route path="/" element={<HomeScreen />} />
                        <Route path="/about" element={<AboutScreen />} />
                        <Route path="/login" element={<LoginScreen />} />
                        <Route path='*' element={<Navigate to='/' />} />
                        {/* <Redirect to='/'/> */}
                    {/* </Switch> */}
                    </Routes>
                </div>

            </div>
        </Router>
    )
}
