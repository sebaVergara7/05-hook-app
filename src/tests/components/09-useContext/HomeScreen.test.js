import React from 'react';
import { mount } from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en HomesScreen.test.js', () => {

    const user = {
        name: 'Sebasti√°n',
        email: 'seba@gmail.com',
    }

    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen /> 
        </UserContext.Provider>
    );

    test('debe de mostrarse correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });

});