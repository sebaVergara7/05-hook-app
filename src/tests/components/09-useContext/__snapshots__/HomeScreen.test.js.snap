// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en HomesScreen.test.js debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Sebasti√°n",
   "email": "seba@gmail.com"
}
  </pre>
</div>
`;
